python/tank/util/yaml_cache.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) 2015 Shotgun Software Inc.
```

Comment:
Couldn't the original code be decorated instead so you wouldn't have to update every single place where we are reading the yam files?